### PR TITLE
`sage.matroids`: Update `# needs`, modularization fixes for imports

### PR DESCRIPTION
**Pull request by mkoeppe**
### :books: Description

- Many `# optional` are removed because small prime finite fields no longer need to be marked `# optional/needs sage.rings.finite_rings` since #35847 
- Other `# optional` are replaced by codeblock-scoped `# needs` tags
- Remaining `# optional` with standard features are replaced by `# needs`
- Some import fixes needed for separate testing of **sagemath-modules** or **sagemath-graphs**[modules] from #35095

This is:
- done with the help of `sage -fixdoctests` from #35749
- Part of: #29705 
- Cherry-picked from: #35095 

### :memo: Checklist

- [x] The title is concise, informative, and self-explanatory.
- [x] The description explains in detail what this PR is about.
- [x] I have linked a relevant issue or discussion.
- [ ] I have created tests covering the changes.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/matroids/dual_matroid.py [hunk 1]
```diff
@@ -487,17 +487,17 @@ def __ne__(self, other):
...
-            sage: M1 = matroids.named_matroids.Fano()                                   # optional - sage.rings.finite_rings
-            sage: M2 = CircuitClosuresMatroid(M1.dual())                                # optional - sage.rings.finite_rings
-            sage: M3 = CircuitClosuresMatroid(M1).dual()                                # optional - sage.rings.finite_rings
+            sage: M1 = matroids.named_matroids.Fano()
+            sage: M2 = CircuitClosuresMatroid(M1.dual())
+            sage: M3 = CircuitClosuresMatroid(M1).dual()
             sage: M4 = CircuitClosuresMatroid(groundset='abcdefg',
             ....:   circuit_closures={3: ['abcdefg'], 2: ['beg', 'cdb', 'cfg',
             ....:                'ace', 'fed', 'gad', 'fab']}).dual()
-            sage: M1.dual() != M2  # indirect doctest                                   # optional - sage.rings.finite_rings
```
Comment (dcoudert) [inline src/sage/matroids/dual_matroid.py:496]: Why removing the `indirect doctest` statement ? Isn't it useful ?

**Review by dcoudert — CHANGES_REQUESTED** [review]
This is much lighter this way, in particular with the use of block statements like `sage: # needs sage.graphs`.

File: src/sage/matroids/dual_matroid.py (hunk 1 above)
Comment (mkoeppe) [reply src/sage/matroids/dual_matroid.py:496]: Thanks for catching this mistake. I've restored it in 4d112691ff

**Review by dcoudert — APPROVED** [review]
LGTM.

**Comment by mkoeppe** [comment]
Thank you!

**Pull request merged**